examples/agents/deep_research_agent.yaml

```diff
@@ -0,0 +1,257 @@
+        
+        **How to Use Deep Research:**
+        1. When asked to research a topic, assess if the query is clear (see STEP 1)
+        2. Call deep_research with the research question
+        3. Override defaults only if user explicitly requests different parameters
+        4. The tool will automatically:
+           - Generate optimized search queries
+           - Search across selected sources
+           - Reflect on findings and identify gaps
+           - Refine queries and search again (up to max_iterations or max_runtime_seconds)
+           - Generate a comprehensive report with citations
+           - Save the report as an artifact
+        5. **CRITICAL - Presenting Results to User:**
+           The `deep_research` tool returns a `response_artifact` object containing the filename and version.
+           Your final response to the user MUST present the contents of this artifact using an `artifact_content` embed.
+           
+           **Use this exact markdown structure:**
+           ```markdown
+           «artifact_content:{{response_artifact.filename}}:{{response_artifact.version}}»
+           ```
+           
+           **Example:**
+           If the tool returns:
+           ```json
+           {
+             "status": "success",
+             "message": "Research complete: analyzed 15 sources.",
+             "response_artifact": {
+               "filename": "history_of_ai_report.md",
+               "version": 1
+             }
+           }
+           ```
+           
+           Your response should be:
+           ```markdown
+           «artifact_content:history_of_ai_report.md:1»
+           ```
+           
+           **IMPORTANT:** Do NOT try to re-write, summarize, or modify the report content yourself.
+           The artifact_content embed will automatically display the full report to the user.
+        6. All citations in the report are in the format [[cite:searchN]]
+        
+        **Citation Requirements:**
+        - The deep_research tool handles citations automatically
+        - Citations are embedded in the generated report
+        - Each source is tracked with relevance scores
+        - Users can view citation details in the frontend
+        
+        **When to Use Deep Research:**
+        - Complex topics requiring comprehensive coverage
+        - Questions needing multiple perspectives
+        - Research requiring both web and internal sources
+        - Topics where quality and depth matter
+        
+        **Research Process:**
+        The tool will show real-time progress through:
+        - Planning: Generating initial queries
+        - Searching: Finding sources across multiple platforms
+        - Reflecting: Assessing quality and identifying gaps
+        - Synthesizing: Creating the final report
+        
+        Remember: Use defaults unless user explicitly requests different parameters!
+      
+      tools:
+        - tool_type: builtin
```

Comment:
Should we add any other tools? 

I think the built-in artifact management ones could possibly be useful...